tests/core/test_serializable.py

```diff
@@ -0,0 +1,500 @@
+import re
+
+import pytest
+
+from ssz import (
+    decode,
+    encode,
+)
+from ssz.exceptions import (
+    DeserializationError,
+    SerializationError,
+)
+from ssz.sedes import (
+    List,
+    Serializable,
+    bytes_sedes,
+    uint32,
+    uint32_list,
+)
+from ssz.utils import (
+    infer_sedes,
+)
+
+
+class SSZType1(Serializable):
+    fields = [
+        ('field1', uint32),
+        ('field2', bytes_sedes),
+        ('field3', uint32_list)
+    ]
+
+
+class SSZType2(Serializable):
+    fields = [
+        ('field2_1', SSZType1),
+        ('field2_2', List(SSZType1)),
+    ]
+
+
+class SSZType3(Serializable):
+    fields = [
+        ('field1', uint32),
+        ('field2', uint32),
+        ('field3', uint32),
+    ]
+
+    def __init__(self, field2, field1, field3, **kwargs):
+        super().__init__(field1=field1, field2=field2, field3=field3, **kwargs)
+
+
+class SSZType4(SSZType3):
```

Comment:
Can we add another test class with two `uint32` fields? Then we can have a test to check that this one cannot deserialize `SSZType3` and the other way around (related to the potential container end index bug I mentioned in the comment regarding `Serializable.deserialize_segment`).

Reply:
It's working fine after I added support for classes which are declared with `empty fields` (reference https://github.com/ethereum/pyrlp/issues/101). 

Attaching the screenshot for reference.
![screenshot from 2018-12-24 11-34-23](https://user-images.githubusercontent.com/15922456/50391970-17e31380-0770-11e9-9c79-d70994c39179.png)